### PR TITLE
Added per route initcwnd/initrwnd option

### PR DESCRIPTION
**Pull request by n3wtype**
## Description

Add possibility to set initcwnd/initrwnd options on routes.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:581]:
```suggestion
    ``initcwnd`` (scalar) – since **0.102**
```
This would most probably go into 0.102

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:583]:
```suggestion
          Must be a positive integer value.
```
Fix indentation.

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:585]:
```suggestion
    ``initrwnd`` (scalar) – since **0.102**
```
This would most probably go into 0.102.

File: src/networkd.c
Comment (slyon) [inline src/networkd.c:450]: Please fix indentation of this block to 4 spaces (instead of 5).

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:589]: Please remove one of the empty lines.

File: src/parse.c
Comment (slyon) [inline src/parse.c:1522]:
```suggestion
    {"initrwnd", YAML_SCALAR_NODE, handle_routes_guint, NULL, route_offset(initrwnd)},
```
Please fix trailing whitespace.

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:586]:
```suggestion
     :    The initrwnd to be used for the route, represented by number of segments. 
```
Fix typo.

File: doc/netplan.md
Comment (slyon) [inline doc/netplan.md:582]:
```suggestion
     :    The initcwnd to be used for the route, represented by number of segments.
```
Fix typo.

File: tests/integration/routing.py
Comment (slyon) [inline tests/integration/routing.py:211]: Please fix indentation of `routes:` block.

**Review by slyon — APPROVED** [review]
Thank you very much for your contribution!
This looks like a nice pull request, containing unit- and integration tests from the start. Nice work!

Before we get this merged, I'd like to see some smaller typo and formatting fixes (which I commented on inline), run it through the full integration testing suite and have someone else have another look at the schema changes for the new `initcwnd` and `initrwnd` settings – maybe @vorlonofportland ?

Otherwise this LGTM!

File: tests/integration/routing.py
Comment (slyon) [inline tests/integration/routing.py:224]:
```suggestion
        - 192.168.5.99/24
```
Just a tiny nitpick I found on the 2nd look.

**Review by slyon — APPROVED** [review]
Thanks a lot for your updates!
I ran the unit- & integration tests locally and can confirm everything is working as expected. This LGTM :eyes: now from a code & test perspective – Just one more tiny nitpick (whitespace formatting), which I commented inline.

Let's wait for the schema/naming review before we merge it.

**Comment by cztk** [comment]
Awesome work, thank you!
Came across this patch and can't wait to have this feature as dealing with netowrkd-dispatcher scripts doesn't work for me and would be a mess anyway.

File: doc/netplan.md
Comment (vorlonofportland) [inline doc/netplan.md:585]: Suggest naming this advertised-receive-window (same rationale).

**Review by vorlonofportland — CHANGES_REQUESTED** [review]

**Comment by slyon** [comment]
Thank you very much Steve for you recommendation wrt. the naming of the new options.
I've renamed the settings & variables accordingly and rebased the branch on top of current `master`.

This is now ready for merging.
Thank you very much @n3wtype for your contribution!

**Pull request merged**
